src/node.rs

```diff
@@ -705,106 +833,125 @@ impl EthNamespaceT for InMemoryNode {
             Option<zksync_types::api::Block<zksync_types::api::TransactionVariant>>,
         >,
     > {
-        // Currently we support only hashes for blocks in memory
-        let reader = self.inner.read().unwrap();
-        let not_implemented_format = format!("get_block_by_hash__{}", hash);
+        let inner = Arc::clone(&self.inner);
 
-        let matching_transaction = reader.tx_results.get(&hash);
-        if matching_transaction.is_none() {
-            return not_implemented(&not_implemented_format);
-        }
+        Box::pin(async move {
+            // Currently we support only hashes for blocks in memory
+            let reader = inner
+                .read()
+                .map_err(|_| into_jsrpc_error(Web3Error::InternalError))?;
 
-        let matching_block = reader
-            .blocks
-            .get(&matching_transaction.unwrap().batch_number);
-        if matching_block.is_none() {
-            return not_implemented(&not_implemented_format);
-        }
+            let matching_transaction = reader.tx_results.get(&hash);
+            if matching_transaction.is_none() {
+                return Err(into_jsrpc_error(Web3Error::NotImplemented));
```

Comment:
Same as other comment, I'd like to keep our use of `Web3Error:NotImplemented` wrapped so we can have `println!()` statements